### ISSUE TITLE
Web Interface Never Fully Loads

### ISSUE THREAD

**Issue by jwilczek** [pi-hole/pi-hole#765]
_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**In raising this issue, I confirm the following (please check boxes, eg [X]):**
- [ ] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be _replicated_
- [X] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**
- [X] 1 (very unfamiliar)
- [ ] 2
- [ ] 3
- [ ] 4
- [ ] 5
- [ ] 6
- [ ] 7
- [ ] 8
- [ ] 9
- [ ] 10 (very familiar)

---

**[FEATURE REQUEST | QUESTION | OTHER]:**

_{replace this section with your content or delete if not a FEATURE REQUEST/QUESTION/OTHER}_

**[BUG | ISSUE] Expected Behaviour:**
Web interface should show graphs and statistics.

**[BUG | ISSUE] Actual Behaviour:**
The web interface never finishes fully loading. I only see the indeterminate progress spinning arrows. I've waited for hours and still just the spinning arrows.

**[BUG | ISSUE] Steps to reproduce:**
Happens every time. I'm running latest version and have tried multiple attempts of reinstalling.
## 

**(Optional) Debug Log generated by `pihole -d`:**
http://termbin.com/lxf7

**Comment by Mcat12** [comment]
Please generate a debug log.

**Comment by jwilczek** [comment]
http://termbin.com/lxf7

**Comment by Mcat12** [comment]
Run `du -h /var/log/pihole.log` and `curl -i http://pi.hole/admin/api.php?summary`.

**Comment by Mcat12** [comment]
Is this a fresh system, and is there anything else running on it?

**Comment by jwilczek** [comment]
**du -h /var/log/pihole.log**
6.3M    /var/log/pihole.log

**curl -i http://pi.hole/admin/api.php?summary**
HTTP/1.1 500 Internal Server Error
X-Pi-hole: The Pi-hole Web interface is working!
X-Frame-Options: DENY
Content-type: application/json
Content-Length: 0
Date: Sat, 08 Oct 2016 16:40:37 GMT
Server: lighttpd/1.4.35

Fresh... I've been running pihole on this RPi since the beginning and updating as there are updates. Nothing else is running on this RPI.

**Comment by Mcat12** [comment]
Run `pihole -f` then the curl command again.

**Comment by jwilczek** [comment]
HTTP/1.1 200 OK
X-Pi-hole: The Pi-hole Web interface is working!
X-Frame-Options: DENY
Content-type: application/json
Transfer-Encoding: chunked
Date: Sat, 08 Oct 2016 16:45:54 GMT
Server: lighttpd/1.4.35

{"domains_being_blocked":"104,558","dns_queries_today":"3","ads_blocked_today":"0","ads_percentage_today":"0.0"}

**Comment by Mcat12** [comment]
Try loading the web interface now.

**Comment by jwilczek** [comment]
Everything loaded quickly, except for the "Queries over Time" chart.

**Comment by Mcat12** [comment]
Did that one load eventually? Usually it is the slowest.

**Comment by jwilczek** [comment]
Scratch that... it loaded immediately with a refresh.

So... do I need to periodically flush the log myself?

**Comment by Mcat12** [comment]
No, it will flush every night. We're currently looking into another issue caused by the log, although it might not be related to this one. If it comes up again, reopen this issue.

**Comment by jwilczek** [comment]
Thanks!

**Comment by jwilczek** [comment]
This issue has returned.

**Comment by Slayman** [comment]
I too have the exact same issue. More problematically it's not blocking any ads.
I installed it today (like 2 hours ago)
Raspi 3B, raspbian light

```
HTTP/1.1 200 OK
X-Pi-hole: The Pi-hole Web interface is working!
X-Frame-Options: DENY
Content-type: application/json
Transfer-Encoding: chunked
Date: Wed, 12 Oct 2016 21:12:57 GMT
Server: lighttpd/1.4.35

"domains_being_blocked":"103,935","dns_queries_today":"74","ads_blocked_today":"0","ads_percentage_today":"0.0"}
```

http://termbin.com/mh23

**Comment by dschaper** [comment]
`dnsmasq` does appear to be responding to queries, the debug TXT queries are getting responses and the process is responding to clients, as the large last block of information in the debug shows. It does show that it's responding with the wrong address, can you run `pihole -g` and then do another debug process and post the link? There's a bug that we're trying to track down that often is resolved with `pihole -g`. If that fixes things, let us know and we can add that to the list of items to check in the debugging... Thanks!

**Comment by Slayman** [comment]
Sorry, I did an upgrade (curl -L https://install.pi-hole.net | bash )and that fixes the ad-blocking now:
```
HTTP/1.1 200 OK
X-Pi-hole: The Pi-hole Web interface is working!
X-Frame-Options: DENY
Content-type: application/json
Transfer-Encoding: chunked
Date: Wed, 12 Oct 2016 21:36:33 GMT
Server: lighttpd/1.4.35

{"domains_being_blocked":"103,937","dns_queries_today":"10","ads_blocked_today":"2","ads_percentage_today":"20.0"}
``` it's still, not loading the WebGUI but it's blocking ads as it should.

**Comment by dschaper** [comment]
Can you screenshot the Admin GUI?

**Comment by Slayman** [comment]
https://i.imgur.com/ep0spi9.png
that is what it looks like (the arrow does rotate)

**Comment by dschaper** [comment]
Thanks, we'll take a look and see what we can come up with.

**Comment by Slayman** [comment]
I have reloaded the page without caching again a little after the update I did earlier went through. It not loaded fast and complete. It loads every time now. So the update fixed both issues.

**Comment by aerik** [comment]
My log doesn't seems to be being cleared nightly, causing the same problem.  When the log is short, the web interface works, but when the log grows big the interface quits working.  I actually changed the php code to limit the size of the $log array, but when the file is too big, it still causes problems.  What mechanism is supposed to clear or truncate the log and how do I troubleshoot / fix it?

P.S.  The web interface causes the log to be read several times for one page request - perhaps it's possible to improve so that the log file only has to be scanned once, either by caching (somewhere) the $log array or by only doing one ajax query for all the data necessary?

**Comment by PromoFaux** [comment]
The log should be truncated every night at around midnight on a `cron` job. You can see this in `/etc/cron.d/pihole`.

Basically it just runs `pihole -f` to flush the logs out. (which in turn is just a script that does `echo " " > /var/log/pihole.log`

RE: Performance improvements, we have a couple of things in the works.

This [Pull Request](https://github.com/pi-hole/AdminLTE/pull/156) from @	akoebbe is just waiting on a  final sign off, but essentially does what you're saying...

eventually we hope to go with something like [this pull request](https://github.com/pi-hole/AdminLTE/pull/149) But it's veryvery pre-alpha right now

**Comment by aerik** [comment]
P.P.S.  To fix it in the short term, I'm using this guys php tail function to get the end of the log
http://www.geekality.net/2011/05/28/php-tail-tackling-large-files/

**Comment by my95z34** [comment]
I just set up PiHole yesterday.  It's been running for maybe 24 hours now.  Blocking seems to be working fine but the webui stopped loading yesterday.  Did the above commands seemed to fix it but my log was 19MB.

htop shows that PHP is pegging the CPU hard. http://imgur.com/a/OXVeo

```
pi@raspberrypi:~ $ du -h /var/log/pihole.log
19M     /var/log/pihole.log
pi@raspberrypi:~ $ curl -i http://pi.hole/admin/api.php?summary
HTTP/1.1 500 Internal Server Error
X-Pi-hole: The Pi-hole Web interface is working!
X-Frame-Options: DENY
Content-type: application/json
Content-Length: 0
Date: Sun, 04 Dec 2016 20:45:12 GMT
Server: lighttpd/1.4.35

pi@raspberrypi:~ $ pihole -f
::: Flushing /var/log/pihole.log ...... done!
pi@raspberrypi:~ $ curl -i http://pi.hole/admin/api.php?summary
HTTP/1.1 200 OK
X-Pi-hole: The Pi-hole Web interface is working!
X-Frame-Options: DENY
Content-type: application/json
Transfer-Encoding: chunked
Date: Sun, 04 Dec 2016 20:45:45 GMT
Server: lighttpd/1.4.35

{"domains_being_blocked":"103,818","dns_queries_today":"2","ads_blocked_today":"0","ads_percentage_today":"0.0"}
```

**Comment by DL6ER** [comment]
Are you running a Raspberry Pi version 1? I see similar problems after days with heavy web access...

**Comment by my95z34** [comment]
I am, indeed.

**Comment by DL6ER** [comment]
Note that with a more powerful Raspberry (my other one for Pi-Hole development is a v3) this problem does not exist. It can load even such large log within a few seconds. Very impressive!

There is currently work going on to improve this situation, see e.g. [here](https://github.com/pi-hole/pi-hole/pull/731).

**Comment by compuguy** [comment]
Well since #731 is closed, will this issue be worked on?

**Comment by dschaper** [comment]
https://github.com/pi-hole/FTL

**Comment by compuguy** [comment]
@dschaper Forgot about that.

**Comment by dschaper** [comment]
It's in alpha stage so if you can take a look and see what you think, we'd welcome any feedback on it. But we are very actively developing some new things behind the scenes to see what works out the best.
